### PR TITLE
Implement @copy syntax for quickly getting a string onto the clipboard

### PR DESCRIPTION
**Pull request by gotjoshua**
This is a really handy feature inspired by:
http://brettterpstra.com/projects/quickquestion/#copy-to-clipboard

### CODE REVIEW

**Comment by viddo** [comment]
A bit skeptical about the placement, but I guess that can do for the time being.

Can you fix the CI errors, and add (at least) a happy-case test to have this covered?

**Comment by gotjoshua** [comment]
Any ideas for better placement? 
I just looked for a place that i had access to the content of the currently selected note...

Looks like I fixed the CI linter errors now. 

Sorry, i don't know what you mean by "happy-case test"... 
Can you give me an example and maybe a link to a resource so I could learn how to add one?

Cheers

**Comment by viddo** [comment]
> Any ideas for better placement?
> I just looked for a place that i had access to the content of the currently selected note...

Yeah, I understand. The reason why I'm saying I'm skeptical about the placement is because this side-effect has nothing to do with the preview editor. Also, I just remembered; this method isn't _always_ called for a selection, so might not always work as you intend. It's only when the [preview editor is updated](https://github.com/viddo/atom-textual-velocity/blob/6f156fdac1bcb002b92f34fcd40da52ca6c2c09e/lib/epics/preview-note.js#L62). 

Therefore, I think a more ideal solution would probably to implement a new [epic](https://github.com/viddo/atom-textual-velocity/tree/6f156fdac1bcb002b92f34fcd40da52ca6c2c09e/lib/epics) that encapsulates this side-effect/use-case. I have not tested this but I think this should do:

```js
// epics/at-copy-match-to-clipboard.js
export default function atCopyMatchToClipboardEpic (action$: rxjs$Observable<Action>, store: Store<State, Action>) {
  return action$
    .debounceTime(25) // ms
    .filter(() => {
      const state: State = store.getState()

      if (state.selectedNote) {
        const filename = state.selectedNote.filename
        const note = state.notes[filename]

        if (note && typeof note.content === 'string') {
          const match = content.match(/@copy\(([\S\s]*)\)/)

          if (match && match[1]) {
            atom.clipboard.write(match[1])
          }
        }
      }
      return false
    })
}
```

It would also have to be imported & added in the [epics/index.js](https://github.com/viddo/atom-textual-velocity/blob/master/lib/epics/index.js) to be used.

> Sorry, i don't know what you mean by "happy-case test"...
> Can you give me an example and maybe a link to a resource so I could learn how to add one?

Sorry about that, my bad. I realize I could/should have been clearer. :)

A happy-case test is essentially a test case that covers the typical code path of the use case. In this case that would be asserting that _some-content is written to clipboard when a note with some content contains `@copy(some-content)` is selected.

Many of the existing tests essentially do this, for an example see [preview-notes-pec](https://github.com/viddo/atom-textual-velocity/blob/master/spec/epics/preview-note-spec.js#L90-L105).

The [official docs](http://flight-manual.atom.io/hacking-atom/sections/writing-specs/) is a great resource of how to write and run tests in Atom, in case you're not familiar with them. I'm confident that if you look at that and the existing tests you can figure out how it can be done. :)

**Comment by gotjoshua** [comment]
Cool @viddo, Your code worked with minimal changes needed (@flow and import at the top and note.content.match rather than content.match)

I am not super motivated to make the test case, i must admit. Do you require it before accepting the merge?

**Comment by gotjoshua** [comment]
ps - there is a typo in your post referring to preview-notes-pec instead of  preview-note-spec.
(i will delete this comment when you edit it)

File: lib/epics/at-copy-match-to-clipboard.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/* @flow */
+
+import {Observable} from 'rxjs'
```
Comment (viddo) [inline lib/epics/at-copy-match-to-clipboard.js]: Why is this import necessary?

**Comment by viddo** [comment]
> I am not super motivated to make the test case, i must admit. Do you require it before accepting the merge?

No worries, on second thought I think the flow type checking probably  is enough in this case.

For the future though I'd recommend to write a test case first, that way the motivation is there because it's typically (for logic) way easier to validate implementation against the test than do it manually and it doesn't become a thing one have to do afterwards. :)

File: lib/epics/at-copy-match-to-clipboard.js (hunk 1 above)
Comment (gotjoshua) [inline lib/epics/at-copy-match-to-clipboard.js]: not needed, and removed.

( I ignorantly thought that it would be needed because of this: action$: rxjs$Observable<Action> )

Comment (viddo) [inline lib/epics/at-copy-match-to-clipboard.js]: Not pushed yet?
